### ISSUE TITLE
fine-tune时loss无明显变化

### ISSUE THREAD

**Issue by LinXin04** [FlagOpen/FlagEmbedding#914]
Using the `WANDB_DISABLED` environment variable is deprecated and will be removed in v5. Use the --report_to flag to control the integrations used for logging result (for instance --report_to none).
06/26/2024 16:59:16 - WARNING - __main__ -   Process rank: 0, device: cuda:0, n_gpu: 1, distributed training: True, 16-bits training: True
06/26/2024 16:59:16 - INFO - __main__ -   Training/evaluation parameters RetrieverTrainingArguments(
_n_gpu=1,
accelerator_config={'split_batches': False, 'dispatch_batches': None, 'even_batches': True, 'use_seedable_sampler': True},
adafactor=False,
adam_beta1=0.9,
adam_beta2=0.999,
adam_epsilon=1e-08,
auto_find_batch_size=False,
bf16=False,
bf16_full_eval=False,
colbert_dim=-1,
data_seed=None,
dataloader_drop_last=True,
dataloader_num_workers=0,
dataloader_persistent_workers=False,
dataloader_pin_memory=True,
dataloader_prefetch_factor=None,
ddp_backend=None,
ddp_broadcast_buffers=None,
ddp_bucket_cap_mb=None,
ddp_find_unused_parameters=None,
ddp_timeout=1800,
debug=[],
deepspeed=None,
disable_tqdm=False,
dispatch_batches=None,
do_eval=False,
do_predict=False,
do_train=False,
enable_sub_batch=True,
eval_accumulation_steps=None,
eval_delay=0,
eval_steps=None,
evaluation_strategy=no,
fix_encoder=False,
fix_position_embedding=False,
fp16=True,
fp16_backend=auto,
fp16_full_eval=False,
fp16_opt_level=O1,
fsdp=[],
fsdp_config={'min_num_params': 0, 'xla': False, 'xla_fsdp_v2': False, 'xla_fsdp_grad_ckpt': False},
fsdp_min_num_params=0,
fsdp_transformer_layer_cls_to_wrap=None,
full_determinism=False,
gradient_accumulation_steps=1,
gradient_checkpointing=False,
gradient_checkpointing_kwargs=None,
greater_is_better=None,
group_by_length=False,
half_precision_backend=auto,
hub_always_push=False,
hub_model_id=None,
hub_private_repo=False,
hub_strategy=every_save,
hub_token=<HUB_TOKEN>,
ignore_data_skip=False,
include_inputs_for_metrics=False,
include_num_input_tokens_seen=False,
include_tokens_per_second=False,
jit_mode_eval=False,
label_names=None,
label_smoothing_factor=0.0,
learning_rate=1e-06,
length_column_name=length,
load_best_model_at_end=False,
local_rank=0,
log_level=passive,
log_level_replica=warning,
log_on_each_node=True,
logging_dir=/bge-m3-finetune/runs/Jun26_16-59-16_kmaker-55-033137181170,
logging_first_step=False,
logging_nan_inf_filter=True,
logging_steps=10,
logging_strategy=steps,
lr_scheduler_kwargs={},
lr_scheduler_type=linear,
max_grad_norm=1.0,
max_steps=-1,
metric_for_best_model=None,
mp_parameters=,
neftune_noise_alpha=None,
negatives_cross_device=True,
no_cuda=False,
normlized=True,
num_train_epochs=3.0,
optim=adamw_torch,
optim_args=None,
output_dir=/bge-m3-finetune,
overwrite_output_dir=False,
past_index=-1,
per_device_eval_batch_size=8,
per_device_train_batch_size=4,
prediction_loss_only=False,
push_to_hub=False,
push_to_hub_model_id=None,
push_to_hub_organization=None,
push_to_hub_token=<PUSH_TO_HUB_TOKEN>,
ray_scope=last,
remove_unused_columns=True,
report_to=['tensorboard'],
resume_from_checkpoint=None,
run_name=/bge-m3-finetune,
save_on_each_node=False,
save_only_model=False,
save_safetensors=True,
save_steps=3000,
save_strategy=steps,
save_total_limit=None,
seed=42,
self_distill_start_step=-1,
sentence_pooling_method=cls,
skip_memory_metrics=True,
split_batches=None,
temperature=0.02,
tf32=None,
torch_compile=False,
torch_compile_backend=None,
torch_compile_mode=None,
torchdynamo=None,
tpu_metrics_debug=False,
tpu_num_cores=None,
unified_finetuning=True,
use_cpu=False,
use_ipex=False,
use_legacy_prediction_loop=False,
use_mps_device=False,
use_self_distill=True,
warmup_ratio=0.0,
warmup_steps=0,
weight_decay=0.0,
)
06/26/2024 16:59:16 - INFO - __main__ -   Model parameters ModelArguments(model_name_or_path='/bge-m3/bge-m3/', config_name=None, tokenizer_name=None, cache_dir=None)
06/26/2024 16:59:16 - INFO - __main__ -   Data parameters DataArguments(knowledge_distillation=False, train_data=['./toy_train_data'], cache_path=None, train_group_size=8, query_max_len=512, passage_max_len=128, max_example_num_per_dataset=None, query_instruction_for_retrieval=None, passage_instruction_for_retrieval=None, same_task_within_batch=True, shuffle_ratio=0.0, small_threshold=0, drop_threshold=0)
06/26/2024 16:59:17 - INFO - __main__ -   Config: XLMRobertaConfig {
  "_name_or_path": "/bge-m3/bge-m3/",
  "architectures": [
    "XLMRobertaModel"
  ],
  "attention_probs_dropout_prob": 0.1,
  "bos_token_id": 0,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 1024,
  "id2label": {
    "0": "LABEL_0"
  },
  "initializer_range": 0.02,
  "intermediate_size": 4096,
  "label2id": {
    "LABEL_0": 0
  },
  "layer_norm_eps": 1e-05,
  "max_position_embeddings": 8194,
  "model_type": "xlm-roberta",
  "num_attention_heads": 16,
  "num_hidden_layers": 24,
  "output_past": true,
  "pad_token_id": 1,
  "position_embedding_type": "absolute",
  "torch_dtype": "float32",
  "transformers_version": "4.38.2",
  "type_vocab_size": 1,
  "use_cache": true,
  "vocab_size": 250002
}

06/26/2024 16:59:19 - INFO - FlagEmbedding.BGE_M3.modeling -   loading existing colbert_linear and sparse_linear---------
=========================
Batch Size Dict:
['0-500: 48',
 '500-1000: 32',
 '1000-2000: 20',
 '2000-3000: 18',
 '3000-4000: 14',
 '4000-5000: 14',
 '5000-6000: 12',
 '6000-7000: 10',
 '7000-inf: 8']
=========================
loading data from ./toy_train_data/input.jsonl ...
---------------------------*Rank 0: refresh data---------------------------
/root/miniconda3/lib/python3.10/site-packages/accelerate/accelerator.py:432: FutureWarning: Passing the following arguments to `Accelerator` is deprecated and will be removed in version 1.0 of Accelerate: dict_keys(['dispatch_batches', 'split_batches', 'even_batches', 'use_seedable_sampler']). Please pass an `accelerate.DataLoaderConfiguration` instead: 
dataloader_config = DataLoaderConfiguration(dispatch_batches=None, split_batches=False, even_batches=True, use_seedable_sampler=True)
  warnings.warn(
06/26/2024 16:59:29 - WARNING - accelerate.utils.other -   Detected kernel version 4.9.151, which is below the recommended minimum of 5.5.0; this can cause the process to hang. It is recommended to upgrade the kernel to the minimum version or higher.
{'loss': 0.0358, 'grad_norm': 7.950653076171875, 'learning_rate': 9.996266666666667e-07, 'epoch': 0.0}
{'loss': 0.0943, 'grad_norm': 4.080717086791992, 'learning_rate': 9.991466666666666e-07, 'epoch': 0.0}
{'loss': 0.0712, 'grad_norm': 2.3540778160095215, 'learning_rate': 9.986133333333333e-07, 'epoch': 0.0}
{'loss': 0.0501, 'grad_norm': 3.0830957889556885, 'learning_rate': 9.9808e-07, 'epoch': 0.01}
{'loss': 0.0359, 'grad_norm': 1.7080806493759155, 'learning_rate': 9.975466666666667e-07, 'epoch': 0.01}
{'loss': 0.0483, 'grad_norm': 31.269142150878906, 'learning_rate': 9.970133333333332e-07, 'epoch': 0.01}
{'loss': 0.0233, 'grad_norm': 0.038686126470565796, 'learning_rate': 9.9648e-07, 'epoch': 0.01}
{'loss': 0.026, 'grad_norm': 0.07161810249090195, 'learning_rate': 9.959466666666665e-07, 'epoch': 0.01}
{'loss': 0.0421, 'grad_norm': 14.044439315795898, 'learning_rate': 9.954133333333333e-07, 'epoch': 0.01}
{'loss': 0.0331, 'grad_norm': 3.9582316875457764, 'learning_rate': 9.948799999999999e-07, 'epoch': 0.02}
{'loss': 0.03, 'grad_norm': 3.3202950954437256, 'learning_rate': 9.943466666666666e-07, 'epoch': 0.02}
{'loss': 0.0299, 'grad_norm': 0.3318820297718048, 'learning_rate': 9.938133333333332e-07, 'epoch': 0.02}
{'loss': 0.0523, 'grad_norm': 1.8701430559158325, 'learning_rate': 9.9328e-07, 'epoch': 0.02}
{'loss': 0.3256, 'grad_norm': 0.9888075590133667, 'learning_rate': 9.927466666666667e-07, 'epoch': 0.02}
{'loss': 0.0452, 'grad_norm': 0.11582855880260468, 'learning_rate': 9.922133333333333e-07, 'epoch': 0.02}
{'loss': 0.0779, 'grad_norm': 8.184672355651855, 'learning_rate': 9.9168e-07, 'epoch': 0.03}
{'loss': 0.0359, 'grad_norm': 1.151183843612671, 'learning_rate': 9.911466666666666e-07, 'epoch': 0.03}
{'loss': 0.0332, 'grad_norm': 2.535754680633545, 'learning_rate': 9.906133333333334e-07, 'epoch': 0.03}
{'loss': 0.0231, 'grad_norm': 0.09630654752254486, 'learning_rate': 9.9008e-07, 'epoch': 0.03}
{'loss': 0.0591, 'grad_norm': 1.3166929483413696, 'learning_rate': 9.895466666666667e-07, 'epoch': 0.03}
{'loss': 0.0461, 'grad_norm': 0.2648876905441284, 'learning_rate': 9.890133333333333e-07, 'epoch': 0.03}
{'loss': 0.0305, 'grad_norm': 0.6831297278404236, 'learning_rate': 9.8848e-07, 'epoch': 0.04}
{'loss': 0.04, 'grad_norm': 0.10579819977283478, 'learning_rate': 9.879466666666666e-07, 'epoch': 0.04}
{'loss': 0.0278, 'grad_norm': 0.17120389640331268, 'learning_rate': 9.874133333333333e-07, 'epoch': 0.04}
{'loss': 0.0342, 'grad_norm': 1.5578186511993408, 'learning_rate': 9.8688e-07, 'epoch': 0.04}
{'loss': 0.0436, 'grad_norm': 1.5485690832138062, 'learning_rate': 9.863466666666667e-07, 'epoch': 0.04}
{'loss': 0.0214, 'grad_norm': 1.9746758937835693, 'learning_rate': 9.858133333333332e-07, 'epoch': 0.04}
{'loss': 0.028, 'grad_norm': 3.740877151489258, 'learning_rate': 9.8528e-07, 'epoch': 0.04}
{'loss': 0.05, 'grad_norm': 15.765501976013184, 'learning_rate': 9.847466666666665e-07, 'epoch': 0.05}
{'loss': 0.0245, 'grad_norm': 0.09108269959688187, 'learning_rate': 9.842133333333333e-07, 'epoch': 0.05}
{'loss': 0.0269, 'grad_norm': 3.434258222579956, 'learning_rate': 9.836799999999999e-07, 'epoch': 0.05}
{'loss': 0.1074, 'grad_norm': 1.7728034257888794, 'learning_rate': 9.831466666666666e-07, 'epoch': 0.05}
{'loss': 0.0254, 'grad_norm': 1.5296252965927124, 'learning_rate': 9.826133333333332e-07, 'epoch': 0.05}
{'loss': 0.0257, 'grad_norm': 2.175490379333496, 'learning_rate': 9.8208e-07, 'epoch': 0.05}
{'loss': 0.1994, 'grad_norm': 1.0658822059631348, 'learning_rate': 9.816e-07, 'epoch': 0.06}
{'loss': 0.1455, 'grad_norm': 2.988440752029419, 'learning_rate': 9.810666666666666e-07, 'epoch': 0.06}
{'loss': 0.0407, 'grad_norm': 0.012402284890413284, 'learning_rate': 9.805333333333334e-07, 'epoch': 0.06}
{'loss': 0.0331, 'grad_norm': 0.003508088644593954, 'learning_rate': 9.8e-07, 'epoch': 0.06}
{'loss': 0.0309, 'grad_norm': 0.5532658696174622, 'learning_rate': 9.794666666666667e-07, 'epoch': 0.06}
{'loss': 0.0399, 'grad_norm': 17.069812774658203, 'learning_rate': 9.789333333333333e-07, 'epoch': 0.06}
{'loss': 0.0815, 'grad_norm': 1.9076682329177856, 'learning_rate': 9.784e-07, 'epoch': 0.07}
{'loss': 0.0475, 'grad_norm': 0.1752307116985321, 'learning_rate': 9.778666666666666e-07, 'epoch': 0.07}
{'loss': 0.0298, 'grad_norm': 0.002476870082318783, 'learning_rate': 9.773333333333333e-07, 'epoch': 0.07}
{'loss': 0.0351, 'grad_norm': 0.10623569786548615, 'learning_rate': 9.768e-07, 'epoch': 0.07}
{'loss': 0.0223, 'grad_norm': 0.038751497864723206, 'learning_rate': 9.762666666666667e-07, 'epoch': 0.07}
{'loss': 0.0396, 'grad_norm': 3.571157455444336, 'learning_rate': 9.757333333333332e-07, 'epoch': 0.07}
{'loss': 0.0268, 'grad_norm': 2.9945828914642334, 'learning_rate': 9.752e-07, 'epoch': 0.08}
{'loss': 0.0291, 'grad_norm': 0.16733404994010925, 'learning_rate': 9.746666666666666e-07, 'epoch': 0.08}
{'loss

**Comment by staoxiao** [comment]
toy_train_data只是样咧数据，模型曾经训练过这些数据，不会有太大变化。

**Comment by LinXin04** [comment]
toy_train_data里面的数据已经替换成了我自己的领域数据，这个loss解决了，但是微调后发现有的打分是负值
print("new_model_score:",predict('Luisa Spagnoli Spa','Replay Residence',model_new))
{'colbert': [-0.006386985071003437], 'sparse': [0.0], 'dense': [0.005908966064453125], 'sparse+dense': [0.005908966064453125], 'colbert+sparse+dense': [-0.00023900950327515602]}

**Comment by LinXin04** [comment]
@staoxiao

**Comment by staoxiao** [comment]
colbert打分理论也不应该存在负值，请问有设置normlize=True吗

**Comment by LinXin04** [comment]
!torchrun --nproc_per_node 1\
-m FlagEmbedding.BGE_M3.run \
--output_dir /bge-m3-finetune \
--model_name_or_path /bge-m3/bge-m3/ \
--train_data ./toy_train_data \
--learning_rate 2e-5 \
--fp16 \
--num_train_epochs 10 \
--per_device_train_batch_size 32 \
--dataloader_drop_last False \
--normlized True \
--temperature 0.2 \
--query_max_len 128 \
--passage_max_len 256 \
--train_group_size 8 \
--negatives_cross_device False\
--logging_steps 30 \
--save_steps 3000 \
--fix_encoder False \
--same_task_within_batch True \
--unified_finetuning True \
--use_self_distill False
@staoxiao

**Comment by LinXin04** [comment]
!torchrun --nproc_per_node 1
-m FlagEmbedding.BGE_M3.run
--output_dir /bge-m3-finetune
--model_name_or_path /bge-m3/bge-m3/
--train_data ./train_data
--learning_rate 2e-5
--fp16
--num_train_epochs 10
--per_device_train_batch_size 32
--dataloader_drop_last False
--normlized True
--temperature 0.2
--query_max_len 128
--passage_max_len 256
--train_group_size 8
--negatives_cross_device False
--logging_steps 30
--save_steps 3000
--fix_encoder False
--same_task_within_batch True
--unified_finetuning True
--use_self_distill False
@staoxiao
